builder/builder.go

```diff
@@ -165,25 +165,34 @@ func (b *Builder) Boot(ctx context.Context) (bool, error) {
 
 	baseCtx := ctx
 	eg, _ := errgroup.WithContext(ctx)
+	errCh := make(chan error, len(toBoot))
+	var mu sync.Mutex
 	for _, idx := range toBoot {
 		func(idx int) {
 			eg.Go(func() error {
 				pw := progress.WithPrefix(printer, b.NodeGroup.Nodes[idx].Name, len(toBoot) > 1)
 				_, err := driver.Boot(ctx, baseCtx, b.nodes[idx].Driver, pw)
 				if err != nil {
 					b.nodes[idx].Err = err
+					mu.Lock()
+					errCh <- err
+					mu.Unlock()
```

Comment:
Is the mutex here needed? I don't think this requires a mutex because `errCh` is written to outside of the area where it's used and it doesn't seem to be changed or modified until after the error group is finished.

Also `errCh` is initialized with a buffer the size of the number of nodes to boot so this should be non-blocking and a-ok without any other code.

Reply:
Ah my bad yes there should be no contention, and it won't block until the buffer is full.